pkg/target/partition.go

```diff
@@ -37,7 +37,7 @@ func manualPartition(rollout *fleet.RolloutStrategy, targets []*Target) ([]Parti
 	targetLoop:
 		for _, target := range targets {
 			for _, cg := range target.ClusterGroups {
-				if matcher.Match(cg.Name, cg.Labels, target.Cluster.Labels) {
+				if matcher.Match(target.Cluster.Name, cg.Name, cg.Labels, target.Cluster.Labels) {
 					partitionTargets = append(partitionTargets, target)
 					continue targetLoop
```

Comment:
Ah, so is _this_ why we need the cluster name?

Reply:
Yeah, I am adding clusterName support because
1. An alternative way to scheduling bundle to just one cluster without dealing with labels.
2. To support fleet-agent bundle per cluster so that a bundle can schedule exact one cluster without dealing with labels